File: examples/C++/MyProjects/Doodle/step04/step04.cc
// ============================================================================
// step01.cc: draw a single squiggle in the main window
//   click the left mouse & drag around the window to draw squiggle/doodle.
//
// Tutorial - Qt Scribble Application
// Based on a similar tutorial for Borland ObjectWindows Library (OWL)
//
// @author Manish Bhobe for Nämostuté Ltd.
// My experiments with C++,Qt, Python & PyQt.
// Code is provided for illustration purposes only! Use at your own risk.
// =============================================================================

#include "DrawWindow.h"
#include "chocolaf.h"
#include <QApplication>
#include <QtGui>

const QString AppTitle("Qt Scribble");

int main(int argc, char** argv)
{
    Chocolaf::ChocolafApp::setupForHighDpiScreens();
    // Chocolaf::ChocolafApp app(argc, argv);
    QApplication app(argc, argv);
    app.setStyle("Fusion");
    app.setApplicationName(app.translate("main", AppTitle.toStdString().c_str()));

    // create the GUI
    DrawWindow mainWindow;
    Chocolaf::centerOnScreenWithSize(mainWindow, 0.75, 0.75);
    // mainWindow.resize(QGuiApplication::primaryScreen()->availableSize() * 4 / 5);
    mainWindow.show();

    return app.exec();
}

File: examples/Python/MyProjects/PyDoodle/step04/step04.py
"""
// ================================================================================
// step04.py: drawing a squiggle in the main window
//  application handles left mouse press & drag, right mouse press events
//   - draws a squiggle in the client window when left mouse is pressed & dragged
//   - erases the squiggle when the right mouse is pressed
//
// Tutorial - PyQt5 Doodle Application
// Based on a similar tutorial for Borland ObjectWindows Library (OWL)
// @author: Manish Bhobe
// My experiments with the Qt Framework. Use at your own risk!!
// =================================================================================
"""
import sys
from mainWindow import *

import chocolaf


def main():
    chocolaf.enable_hi_dpi()
    # app = ChocolafApp(sys.argv)
    # app.setStyle("WindowsDark")
    app = QApplication(sys.argv)
    app.setStyle("Fusion")

    mainWindow = MainWindow()
    mainWindow.show()

    sys.exit(app.exec())


if __name__ == "__main__":
    main()
